lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
@Iristyle have you talked me out of this implementation? This is public API, and more knobs without reason suck. @adrienthebo, would a whitelist style implementation suck? Can we do something like this, and make it private I wonder? Being able to mark idempotency issues for other custom types really just fell out of this effort, its not essential to success (accidental feature?), at least I'm not entirely sure it is. We can always add it in, when its called for?